compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -507,8 +507,9 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         span: Span,
     ) -> LocalDefId {
         debug_assert_ne!(node_id, ast::DUMMY_NODE_ID);
-        assert!(
-            self.opt_local_def_id(node_id).is_none(),
+        debug_assert_eq!(
+            self.opt_local_def_id(node_id),
```

Comment:
Not very hot (only called for definitions created by syntactic sugar), but the assert performs a non-trivial calculation.
Not sure about usefulness of this assert.